File: lab1/src/main/java/Validator/ReservationValidator.java
package Validator;

import Domain.Reservation;
import Errors.ValidationError;

public class ReservationValidator implements ValidatorInterface<Reservation> {
  @Override
  public void Validate(Reservation entity) throws ValidationError {
    String errors = "";
    if(entity.getSeatsNr() < 0 || entity.getSeatsNr() > 18)
      errors = errors.concat("You can't reserve that many seats!\n");
    if(entity.getClientName().equals(""))
      errors = errors.concat("Invalid client name!");
    if(!errors.equals(""))
      throw new ValidationError(errors);
  }
}

File: proiect_c#/mpp_csharp/src/Validator/ReservationValidator.cs

public class ReservationValidator: IValidator<Reservation> {
  public void Validate(Reservation entity) {
    string errors = "";
    if(entity.SeatsNr < 0 || entity.SeatsNr > 18)
      errors += "You can't reserve that many seats!\n";
    if(entity.ClientName.Equals(""))
      errors += "Invalid client name!";
    if(!errors.Equals(""))
      throw new ValidationError(errors);
  }
}
